src/main/java/com/google/starfish/servlets/SearchServlet.java

```diff
@@ -0,0 +1,81 @@
+package com.google.starfish.servlets;  
+
+import java.io.IOException;  
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import javax.servlet.ServletException;  
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import java.util.Date;
+import com.google.starfish.services.NoteService;
+import com.google.starfish.models.Note;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
+/** Servlet that returns search results for notes based on school and course */
+@WebServlet("/search")  
+public class SearchServlet extends HttpServlet {  
+
+  private NoteService noteService = new NoteService();
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
+    String reqSchool = req.getParameter("school").toLowerCase();
+    String reqCourse= req.getParameter("course").toLowerCase();
+    ArrayList<Note> notes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      String stmt = 
+          "SELECT * "
+        + "FROM `starfish.notes` "
+        + "WHERE `school`= ? AND `course` = ?;";
```

Comment:
I'm not super familiar with SQL but would this statement be able to handle the case that either `school` or `course` might be an empty string?

Reply:
This is a good question, and I think the answer is no, which might not be exactly what we want. I presume that what should happen if one of school or course is an empty string should be we get all notes with that school or course irrespective of what the empty string field is. Right now, if school or course were empty string, we would probably not get back any notes because right now the query is looking for exact matches. This also makes me think of the possibility of extra whitespace on the search query so I'll push a change that handles that. 
I'm not sure if the empty string problem could be handled without having different querries depending on the case so for now I'll add some conditional logic that changes the statement based on what's empty. If we think of a more efficient way to do this (maybe simply through SQL syntax) we could add that later.

Reply:
Alright I found [this](https://stackoverflow.com/questions/50110968/sql-query-handling-multiple-where-conditions-with-a-potential-null-value) which seems to suggest that the two basic approaches to this problem are dynamic query generation (what I proposed to do with conditional logic) or handle NULL checks in a static SQL statement. Since there aren't many variables I'll use dynamic query generation but in the future if a similar problem arises and the query is more complex, it would make sense to consider handling NULL with a static query.

Reply:
Ok awesome! Also, just letting you know that the search bar component also trims the whitespace. Do you think that should be be removed then if it's redundant?

Reply:
That's good to know! I think we'll leave it in just for safety. I don't think it hurts having those checks on both the frontend and backend.

Reply:
Great conversation.